### PR TITLE
Sharing recipient creation 

### PR DESCRIPTION
**Pull request by Gara64**
This allows to create a sharing recipient throught a `POST sharings/recipient` route. 

It triggers the creation of the recipient in the sharer database, and the OAuth registration of the sharer to the Cozy's recipient.
Note that the` public_name` option in the instance settings has been added, as it is needed for the OAuth `client_name` parameter.

### CODE REVIEW

File: pkg/instance/instance.go [hunk 1]
```diff
@@ -134,8 +135,9 @@ func (i *Instance) Included() []jsonapi.Object {
 
 // settings is a struct used for the settings of an instance
 type instanceSettings struct {
-	Timezone string `json:"tz,omitempty"`
-	Email    string `json:"email,omitempty"`
+	Timezone   string `json:"tz,omitempty"`
+	Email      string `json:"email,omitempty"`
+	PublicName string `json:"public_name,omitempty"`
```
Comment (jinroh) [inline pkg/instance/instance.go]: Would be interesting to add the possibility to specify this public name in the:
  - `/instance` API (defined in `web/instance` via a `?PublicName` query param)
  - in the client (`client/instances#CreateInstance` via a `PublicName` field in the `InstanceOptions` struct)
  - in the CLI (`cmd/instances#addInstanceCmd`, via a `--public-name` flag for instance)

Comment (Gara64) [inline pkg/instance/instance.go]: Agreed, but I think it should be another PR to avoid to make this one too big.

Comment (jinroh) [inline pkg/instance/instance.go]: @Gara64 no problem

File: web/sharings/sharings_test.go [hunk 2]
```diff
@@ -187,21 +210,29 @@ func TestCreateSharingSuccess(t *testing.T) {
...
 	}
-
 	clientOAuth, _ = setup.GetTestClient("")
 	clientID = clientOAuth.ClientID
 
 	ts = setup.GetTestServer("/sharings", Routes)
+	ts2 = setup2.GetTestServer("/auth", auth.Routes)
+
 	os.Exit(setup.Run())
+	os.Exit(setup2.Run())
```
Comment (aenario) [inline web/sharings/sharings_test.go]: setup2 will never run, (os.Exit closes the process), so the second Instance will not be cleaned up.
It is a bit ugly, but you can add a public Cleanup to test_utils
```
func (c *TestSetup) Cleanup() {
	c.cleanup()
}
```

and then link the two testSetup
```
setup.AddCleanup(setup2.Cleanup)
```
